lib/open_food_network/option_value_namer.rb

```diff
@@ -63,17 +62,34 @@ def option_value_value_unit_scaled
     end
 
     def scale_for_unit_value
-      units = { 'weight' => { 1.0 => 'g', 1000.0 => 'kg', 1_000_000.0 => 'T' },
-                'volume' => { 0.001 => 'mL', 1.0 => 'L',  1000.0 => 'kL' } }
-
-      # Find the largest available unit where unit_value comes to >= 1 when expressed in it.
+      units = {
+        'weight' => {
+          1.0 =>         { 'name' => 'g',  'system' => 'metric' },
+          28.35 =>       { 'name' => 'oz', 'system' => 'imperial' },
+          453.6 =>       { 'name' => 'lb', 'system' => 'imperial' },
+          1000.0 =>      { 'name' => 'kg', 'system' => 'metric' },
+          1_000_000.0 => { 'name' => 'T',  'system' => 'metric' }
+        },
+        'volume' => {
+          0.001 =>  { 'name' => 'mL', 'system' => 'metric' },
+          1.0 =>    { 'name' => 'L', 'system' => 'metric' },
+          1000.0 => { 'name' => 'kL', 'system' => 'metric' }
+        }
+      }
+
+      # Find the largest available and compatible unit where unit_value comes
+      # to >= 1 when expressed in it.
       # If there is none available where this is true, use the smallest available unit.
-      unit = units[@variant.product.variant_unit].select { |scale, _unit_name|
-        @variant.unit_value / scale >= 1
-      }.to_a.last
-      unit = units[@variant.product.variant_unit].first if unit.nil?
-
-      unit
+      scales = units[@variant.product.variant_unit]
+      product_scale = @variant.product.variant_unit_scale
+      product_scale_system = scales[product_scale.to_f]['system']
+      largest_unit = scales.select { |scale, unit_info|
+        unit_info['system'] == product_scale_system &&
+          @variant.unit_value / scale >= 1
+      }.max
+      largest_unit = units[@variant.product.variant_unit].first if largest_unit.nil?
```

Comment:
this could be
   `   largest_unit = scales.first if largest_unit.nil?`

Reply:
I made this method easier to read. please see my PR.

Reply:
Thanks, I saw that in the PR, it looks good.